### PR TITLE
Add updateOperatorWithSignature

### PR DESCRIPTION
**Pull request by kbrizzle**
Adds a second version of `MarketFactory.updateOperator()` that takes a signed message from the account instead of assuming the sender is the account.

### CODE REVIEW

File: packages/perennial-verifier/contracts/Verifier.sol [hunk 1]
```diff
@@ -75,6 +76,18 @@ contract Verifier is IVerifier, EIP712 {
...
+    /// @notice Verifies the signature of a operator approval type
+    /// @dev Cancels the nonce after verifying the signature
+    /// @param operatorUpdate The operator approval message to verify
+    /// @param signature The signature of the account for the operator approval
+    /// @return The address corresponding to the signature
+    function verifyOperatorUpdate(OperatorUpdate calldata operatorUpdate, bytes calldata signature)
+        external
+        validateAndCancel(operatorUpdate.common, signature) returns (address)
+    {
+        return ECDSA.recover(_hashTypedDataV4(OperatorUpdateLib.hash(operatorUpdate)), signature);
```
Comment (EdNoepel) [inline packages/perennial-verifier/contracts/Verifier.sol:88]: I read up on this at https://docs.openzeppelin.com/contracts/5.x/api/utils#EIP712-_hashTypedDataV4-bytes32- .
Glad there's an OZ library which handles this for us.

File: packages/perennial-verifier/contracts/types/OperatorUpdate.sol [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+using OperatorUpdateLib for OperatorUpdate global;
+
+/// @title OperatorUpdateLib
+/// @notice Library for OperatorUpdate logic and data.
+library OperatorUpdateLib {
+    bytes32 constant public STRUCT_HASH = keccak256("OperatorUpdate(address operator,bool approved,Common common)Common(address account,address domain,uint256 nonce,uint256 group,uint256 expiry)");
+
+    function hash(OperatorUpdate memory self) internal pure returns (bytes32) {
+        return keccak256(abi.encode(STRUCT_HASH, self.operator, self.approved,CommonLib.hash(self.common)));
```
Comment (EdNoepel) [inline packages/perennial-verifier/contracts/types/OperatorUpdate.sol:26]: This jives with OZ's documentation on how to hash the message.  
nit: missing whitespace before `CommonLib.hash`

**Review by arjun-io — APPROVED** [review]

**Review by EdNoepel — APPROVED** [review]
LGTM

File: packages/perennial/contracts/MarketFactory.sol [hunk 3]
```diff
@@ -56,8 +62,26 @@ contract MarketFactory is IMarketFactory, Factory {
...
     function updateOperator(address operator, bool newEnabled) external {
-        operators[msg.sender][operator] = newEnabled;
-        emit OperatorUpdated(msg.sender, operator, newEnabled);
+        _updateOperator(msg.sender, operator, newEnabled);
+    }
+
+    /// @notice Updates the status of an operator for the signer verified via a signed message
+    /// @param operatorUpdate The operator update message to process
+    /// @param signature The signature of the operator update message
+    function updateOperatorWithSignature(OperatorUpdate calldata operatorUpdate, bytes calldata signature) external {
```
Comment (EdNoepel) [inline packages/perennial/contracts/MarketFactory.sol:71]: Just thought: Do we want to add this to the interface?

Comment (kbrizzle) [reply packages/perennial/contracts/MarketFactory.sol:71]: Oops, yes we want all functions to be in the interface generally.

File: packages/perennial-verifier/contracts/Verifier.sol (hunk 1 above)
Comment (kbrizzle) [reply packages/perennial-verifier/contracts/Verifier.sol:88]: Yeah this is much easier than it used to be 😅

**Pull request merged**